pypeit/move_median/mmpy.py

```diff
@@ -0,0 +1,59 @@
+from collections import deque
+from bisect import insort, bisect_left
+import itertools
+
+def move_median(seq_pad, window_size):
+    if window_size % 2 == 0:
+        return move_median_even_window(seq_pad, window_size)
+    else:
+        return move_median_odd_window(seq_pad, window_size)
```

Comment:
This is nit-picky ...  Is there any loss in efficiency if you make these two functions the same and put the check for even vs. odd behavior where it's needed?  I.e., there are only two lines that are different between the two functions, and you would change them to:
```
result.append(s[len(d)//2] if odd else (s[len(d)//2] + s[len(d)//2 - 1])/2)
result.append(s[m] if odd else (s[m] + s[m-1])/2)
```
where you would set `odd = window_size % 2 == 0` at the beginning of the function.

Reply:
I agreed with @kbwestfall suggestion here.

Reply:
Good call! I thought that my original way would be better-performing, but I ran some quick performance tests and the difference is at the 0.1% level, so I've consolidated the even/odd behavior into one function.